ccinput/packages/nwchem.py

```diff
@@ -64,34 +139,79 @@ def clean(self, s):
         )
         return "".join([c for c in s if c in WHITELIST])
 
-    def add_option(self, key, option):
-        _option = option.strip()
-        if key in self.commands:
-            if _option not in self.commands[key]:
-                self.commands[key].append(_option)
-        else:
-            self.commands[key] = [_option]
-
-    def add_options(self, key, options):
-        for option in options:
-            self.add_option(key, option)
-
+    def separate_lines(self,text):
+        lines = text.split(';')
+        for line in lines :
+            line = self.clean(line)
+        return "\n".join(lines)
+    
+    def handle_tasks(self):
+        for word in self.KEYWORDS[self.calc.type]:
+            self.tasks += f'task {self.calc.parameters.theory_level} {word} \n'
+        #handle levels of theory
+        if(self.calc.parameters.method == 'hf' or self.calc.parameters.method in SYN_METHODS['hf']) :
+            scf_block = f"""
+            {SOFTWARE_MULTIPLICITY['nwchem'][self.calc.multiplicity]}
+            """
+            self.method_block+= scf_block
+        if(self.calc.parameters.theory_level == 'dft') :
+            dft_block = f"""
+            xc {self.calc.parameters.method}
+            mult {self.calc.multiplicity}
+            """
+            self.method_block += dft_block
+            if self.calc.parameters.d3 :
+                self.method_block += "disp vdw 3 \n"
+            elif self.calc.parameters.d3bj :
+                self.method_block += "disp vdw 4 \n"
+
+    def handle_basis_sets(self):
+        basis_set = get_basis_set(self.calc.parameters.basis_set, "nwchem")
+        if(basis_set != ''):
+            self.basis_set = f"* library {basis_set}"
     def handle_specifications(self):
-        return
-
-    def handle_command(self):
-        return
+            s = self.separate_lines(self.calc.parameters.specifications)
+            for spec in s.split('\n'):
+                if spec.split()[0] not in self.BLOCK_KEYWORDS["method"] and self.calc.type != CalcType.SP :
```

Comment:
With no specification, this line raises an `IndexError` (`spec.split()[0]` is not defined if `spec` has length 0). A couple verifications should be done to handle trivial cases.

Reply:
A simple check if specifications is not an empty string should do the work.